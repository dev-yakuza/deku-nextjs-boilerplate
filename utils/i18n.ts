import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import { i18n as i18nConfig } from '../next-i18next.config'

i18n.use(initReactI18next).init({
  fallbackLng: 'en',
  resources: Object.assign(
    {},
    ...i18nConfig.locales.map((locale) => ({
      [locale]: {
        common: require(`../public/locales/${locale}/common.json`),
      },
    })),
  ),
})

export default i18n
