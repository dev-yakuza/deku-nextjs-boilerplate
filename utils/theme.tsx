import { createTheme } from '@mui/material/styles'

export const theme = createTheme({
  palette: {
    primary: {
      main: '#ff8e88',
    },
  },
  typography: {
    h1: {
      fontSize: '2rem',
    },
  },
})
