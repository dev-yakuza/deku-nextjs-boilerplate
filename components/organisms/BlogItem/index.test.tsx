import { render, screen, fireEvent } from '@testing-library/react'
import { RouterContext } from 'next/dist/shared/lib/router-context'

import { mockGrid, getMockRouter } from 'utils/test'

import { BlogItem } from '.'

describe('<BlogItem />', () => {
  it('Rendered well', async () => {
    const { container } = render(
      <BlogItem
        id={20}
        title="This is the blog title."
        contents="This is the blog contents."
      />,
    )

    const grid = mockGrid.mock.calls[0][0]
    expect(grid.item).toBe(true)
    expect(grid.xs).toBe(12)

    const link = grid.children
    expect(link.type.render.name).toBe('LinkComponent')
    expect(link.props.href).toBe('/posts/20')
    expect(link.props.children.type).toBe('a')

    const titleComponent = link.props.children.props.children[0]
    expect(titleComponent.type.name).toBe('Typography')
    expect(titleComponent.props.children).toBe('This is the blog title.')

    const contentsComponent = link.props.children.props.children[1]
    expect(contentsComponent.type.name).toBe('Typography')
    expect(contentsComponent.props.children).toBe('This is the blog contents.')

    expect(container).toMatchSnapshot()
  })

  it('Go to the post detail page when the blog item is clicked', async () => {
    const routerPush = jest.fn()
    const mockRouter = getMockRouter()
    mockRouter.push = routerPush

    render(
      <RouterContext.Provider value={mockRouter}>
        <BlogItem
          id={40}
          title="This is the blog title."
          contents="This is the blog contents."
        />
      </RouterContext.Provider>,
    )

    expect(routerPush).not.toBeCalled()
    fireEvent.click(screen.getByText('This is the blog title.'))
    expect(routerPush.mock.calls[0][0]).toBe('/posts/40')
  })
})
