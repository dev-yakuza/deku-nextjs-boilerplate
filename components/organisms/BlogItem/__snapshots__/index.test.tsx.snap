// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogItem /> Rendered well 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1h6d3gi-MuiGrid-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      This is the blog title.
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      This is the blog contents.
    </p>
  </div>
</div>
`;
